d2l-image.js

```diff
@@ -16,66 +38,71 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-image">
 				border-radius: var(--d2l-image-border-radius);
 				object-fit: var(--d2l-image-object-fit);
 			}
-		</style>
-
-		<iron-ajax id="imageRequest" url="[[imageUrl]]" headers="[[_headers]]" on-iron-ajax-response="_onImageResponse" on-iron-ajax-error="_onImageResponse" handle-as="blob">
-		</iron-ajax>
-
-		<img id="image" alt$="[[alternateText]]">
-	</template>
-
-</dom-module>`;
-
-document.head.appendChild($_documentContainer.content);
-
-Polymer({
-	is: 'd2l-image',
-
-	properties: {
-		alternateText: {
-			type: String,
-			reflectToAttribute: true,
-		},
-		imageUrl: {
-			type: String,
-			value: null
-		},
-		token: String,
-		_headers: String
-	},
-
-	observers: [
-		'_onImageUrlChange( imageUrl, token )'
-	],
-
-	_onImageUrlChange: function(imageUrl, token) {
-		if (imageUrl && token) {
-			this._headers = {
-				Authorization: 'Bearer ' + token
-			};
-			this.$.imageRequest.generateRequest();
-		} else if (imageUrl) {
-			this.$.image.src = imageUrl;
+		`;
+	}
+
+	constructor() {
+		super();
+		this.imageUrl = undefined;
+		this.token = undefined;
+		this._loadImage();
+	}
+
+	render() {
+		return html`
+			<img alt="${this.altText}" src="${ifDefined(this._imageUrl)}">
+		`;
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('imageUrl') || changedProperties.has('token')) {
+			this._loadImage();
 		}
-	},
+	}
 
-	_onImageResponse: function(response) {
-		if (response.detail.status === 200) {
-			if (this.$.image.src) {
-				URL.revokeObjectURL(this.$.image.src);
+	async _loadImage() {
+		if (!this.imageUrl || !this.token) {
+			return;
+		}
+
+		let response;
+
+		try {
+			const tokenPromise = await (typeof (this.token) === 'function')
+				? this.token()
+				: Promise.resolve(this.token);
+
+			const tokenString = await tokenPromise;
```

Comment:
I might be mistaken, but I think this second await is redundant - even if `this.token()` returns a promise that itself resolves to a promise, (I think) it won't resolve until the nested promise itself resolves... so Line 71 should always itself resolve to a `tokenString`.

Anyway it's not hurting anything, but I think Line 75 isn't necessary.

Reply:
🤔 you definitely might be right.

Admittedly I yanked this code from a different spot (which I also took part in writing). Once the behavior has been solidified (and tested) I'll play around with potentially removing the await

https://github.com/BrightspaceHypermediaComponents/users/blob/23c3a469b3a15b892956659c2853a077c380269d/components/d2l-profile-image-base.js#L287-L310